### PR TITLE
[Challenge 42] Spring boot actuator hiding api key

### LINKED ISSUE #815: Spring Boot Actuator challenge hiding an api key in the audit events
This challenge is about how to not use the spring boot actuator, by hiding an API key in the audit events:
- [ ] Add an `AuditEventRepository`
- [ ] Add an APIkey received event at AuditEventRepository which is randomly generated
- [ ] enable `management.endpoints.web.exposure.include=auditevents` in `application.properties`
- [ ] Create a challenge using the secret at this endpoint and explain why you need to be careful with Actuator configurations

### PR DESCRIPTION
**Pull request by nwolniak**
### What kind of changes does this PR include?

-   [ ] Fixes or refactors
-   [x] A new challenge
-   [ ] Additional documentation
-   [ ] Something else

#### Description

This PR add new challenge based on hiding API key in Spring Boot Actuator audit events.

### Relations

Closes #815 

### Checklist:

-   [x] All the contributions made are solely the work of me and my co-authors
-   [X] I tested the changes in this PR (if applicable)
-   [X] I added unit tests to ensure my change works (when change in Java or on front-end code)
-   [ ] I added UI tests to ensure my UI changes work (when change in the overall UI, not needed if just adding a challenge)
-   [x] The PR passes pre-commit hooks and automated tests

### CODE REVIEW

**Review by commjoen — APPROVED** [review]
Love it! Thank you so much! We are only in the middle of merging a large refactor. Can we hold merging this PR until the refactor is done? Happy to adjust the result and ensure your commits are part of it.

**Comment by commjoen** [comment]
The refactor is completed. Do you want us to adapt your pr to the new code-base? Or would you like to give it a try yourself @nwolniak ?

**Comment by commjoen** [comment]
@nwolniak regarding the issue with compiling: is the lombok processor turned on in the IDE?

**Comment by nwolniak** [comment]
> @nwolniak regarding the issue with compiling: is the lombok processor turned on in the IDE?

@commjoen I have changed java version linked by maven and just didnt restart IDE, so everything is fine

File: src/main/java/org/owasp/wrongsecrets/challenges/docker/Challenge42.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import lombok.extern.slf4j.Slf4j;
+import org.owasp.wrongsecrets.AuditConfiguration;
+import org.owasp.wrongsecrets.challenges.Challenge;
+import org.owasp.wrongsecrets.challenges.Spoiler;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+/** This is a challenge based on finding API Key in Spring Boot Actuator audit events */
+@Slf4j
+@Order(42)
```
Comment (commjoen) [inline src/main/java/org/owasp/wrongsecrets/challenges/docker/Challenge42.java:13]:
```suggestion

```

File: src/main/resources/explanations/challenge42.adoc [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+The developers decided to leverage the power of Spring Boot Actuator
+to monitor and audit their application. The project involved interfacing with various external APIs,
+each requiring a unique key for authentication. The audit events, detailed and informative, seemed like
+the perfect solution for monitoring the inner working of their system.
+
+The team implemented a logging mechanism that included detailed audit events.
+Unbeknownst to them, these logs contained traces of the sacred API keys, hidden in plain sight within
+the audit events.
+
+Can you find this API key?
```
Comment (bendehaan) [inline src/main/resources/explanations/challenge42.adoc:12]:
```suggestion
The developers decided to leverage the power of Spring Boot Actuator
to monitor and audit their application. The project involved interfacing with various external APIs,
each requiring a unique key for authentication. The audit events, detailed and informative, seemed like
the perfect solution for monitoring the inner workings of their system.

The team implemented a logging mechanism that included detailed audit events.
Unbeknownst to them, these logs contained traces of the sacred API key hidden within
the audit events.

Can you find this API key?
```

File: src/main/resources/explanations/challenge42_reason.adoc [hunk 3]
```diff
@@ -0,0 +1,8 @@
+*Why hiding API key in the audit events is not a good idea?*
+
+In a secure and well-architected system, it's crucial to protect sensitive information such as API keys.
+Audit events play a significant role in tracking and monitoring activities within an application.
+
+The API keys should not be visible in plain text in the logs or any other output.
+
+You should always ensure that API keys are not exposed in clear text within the audit logs.
```
Comment (bendehaan) [inline src/main/resources/explanations/challenge42_reason.adoc:8]:
```suggestion
In a secure and well-architected system, protecting sensitive information such as API keys is crucial.
Audit events are significant in tracking and monitoring activities within an application.

Ensure that API keys are not visible in plain text in the (audit) logs or any other output.
```

**Review by bendehaan — COMMENTED** [review]
Thanks for the contribution! Some minor nits in the text.

**Pull request merged**